modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/repo/ModelRepository.java

```diff
@@ -17,13 +17,16 @@
 
 import com.expedia.adaptivealerting.modelservice.entity.Model;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 /**
  * @author kashah
  *
  */
 public interface ModelRepository extends JpaRepository<Model, Integer> {
 
-    Model getModelById(Integer id);
+    @Query("select m.id from Model m where m.modelUUID = :uuid")
+    Integer findIdByModelUUID(@Param("uuid") String uuid);
```

Comment:
Similar comments apply here. I don't see any advantage to projecting the query results to a single ID. Why not simply return the model itself? Same query behind the scenes.